File: product-webapp/webapp/src/app/model/hotel.ts
import { Address } from "./address";
import { Amenties } from "./amenties";
import { Overview } from "./overview";
import { Reservation } from "./reservation";
import { Review } from "./review";
import { Room } from "./room";

export class Hotel {
     hotelName:string="";
       hotelCategory:string="";
 email:string="";
 propertyRules:string[]=[];
  overview:Overview=new Overview();
 amenities:Amenties= new Amenties();
 address:Address=new Address();
 room:Room[]=[];
 review:Review[]=[];
 reservation:Reservation=new Reservation();
}

File: HotelService/src/main/java/com/example/HotelService/model/Hotel.java
package com.example.HotelService.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;
@Document
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Hotel
{
    @Id
private String hotelName;///
private String hotelCategory;//
private String email;
private List<String> propertyRules;
private Overview overview;
private Amenities amenities;
private Address address;
private List<Room> room;
private List<Review> review;


}
